flexmeasures/ui/views/charts.py

```diff
@@ -0,0 +1,129 @@
+from flask_security import roles_accepted
+from bokeh.embed import json_item
+from marshmallow import Schema, fields
+from webargs.flaskparser import use_args
+from flask_cors import cross_origin
+from flask_json import as_json
+
+from flexmeasures.api.v2_0 import flexmeasures_api as flexmeasures_api_v2_0
+from flexmeasures.api.v2_0.routes import v2_0_service_listing
+from flexmeasures.api.common.schemas import PeriodField
+from flexmeasures.data.queries.analytics import get_power_data
+from flexmeasures.ui.views.analytics import make_power_figure
+
+
+"""
+An endpoint to get a power chart.
+
+This will grow to become code for more charts eventually.
+The plan is to separate charts specs from the actual data later,
+and to switch to Altair.
+
+For now, we'll keep this endpoint here, with route and implementation in the same file.
+When we move forward, we'll review the architecture.
+"""
+
+
+v2_0_service_listing["services"].append(
```

Comment:
I don't think this additional service would actually be picked up by the list of services returned by the getService endpoint. You can check with:

`localhost:5000/api/v2_0/getService`

Reply:
Yes it's in there:

```
{
      "access": [
        "admin", 
        "Prosumer"
      ], 
      "description": "Get a Bokeh chart for power data to embed in web pages.", 
      "name": "GET /charts/power"
}
```